ua-snap/hydroviz
# Build netCDFs from source data

This PR assembles netCDF files from the hydroviz source CSV files, and does a basic QC to make sure the values have been assembled correctly. 

To test: 
- Clone the repo somewhere on Chinook, and make sure you have the `snap-geo` environment built.
- Inspect the codebase and see if you can follow the logic of the functions. Can anything be improved here?
- Follow the instructions in the `README.md` to run the code.
- Run the `qc.ipynb` notebook and make sure the tests pass. Examine the netCDF structure, see how this is different than our usual gridded data with XY coords... does this structure make sense, given the data is non-spatial?

**Note that its not necessary to download the files yourself, just use the files in `/import/beegfs/CMIP6/jdpaul3/hydroviz_data` !**

These netCDFs should allow query and visualization, provided we can join them to the vector data in an efficient way. Next step is to attempt to incorporate vector geometry directly into the dataset using [xvec](https://xvec.readthedocs.io/en/latest/intro.html).
